core/remote.go

```diff
@@ -51,83 +51,89 @@ func NewRemoteClient(address string) (Client, error) {
 		open:          true,
 		conn:          nothing{},
 	}
-	go client.loop(address)
-	return client, nil
+	return client, client.loop(address)
 }
 
-func (client *remoteSubscriber) loop(address string) {
+func (client *remoteSubscriber) loop(address string) error {
 	// every time we disconnect, we want to reconnect
```

Comment:
Just FYI, the reason that this was inside the for loop is because these connections will automatically reconnect when disconnected. This remote client is used by the mist replication stuff, and if it doesn't reconnect, then replication can get into a broken state.